# Part 0: Landing Club Loan Data Clean and Format

1. find the NAN data and clean it as necessary
2. format some features to suitable formats for further analysis
3. get a general idea about the data set and content
4. save the data into different group for different analysis purpose

In [3]:
# import the modules
# load the useful package 
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

% matplotlib inline

In [4]:
df = pd.read_csv('inputData/loan.csv',low_memory = False)
df_org = df.copy()
print('data size is :', df.shape)
df.info()
df.head()

data size is : (887379, 74)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887379 entries, 0 to 887378
Data columns (total 74 columns):
id                             887379 non-null int64
member_id                      887379 non-null int64
loan_amnt                      887379 non-null float64
funded_amnt                    887379 non-null float64
funded_amnt_inv                887379 non-null float64
term                           887379 non-null object
int_rate                       887379 non-null float64
installment                    887379 non-null float64
grade                          887379 non-null object
sub_grade                      887379 non-null object
emp_title                      835922 non-null object
emp_length                     887379 non-null object
home_ownership                 887379 non-null object
annual_inc                     887375 non-null float64
verification_status            887379 non-null object
issue_d                        887379 non-null 

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501    1296599     5000.0       5000.0           4975.0   36 months   
1  1077430    1314167     2500.0       2500.0           2500.0   60 months   
2  1077175    1313524     2400.0       2400.0           2400.0   36 months   
3  1076863    1277178    10000.0      10000.0          10000.0   36 months   
4  1075358    1311748     3000.0       3000.0           3000.0   60 months   

   int_rate  installment grade sub_grade     ...      total_bal_il il_util  \
0     10.65       162.87     B        B2     ...               NaN     NaN   
1     15.27        59.83     C        C4     ...               NaN     NaN   
2     15.96        84.33     C        C5     ...               NaN     NaN   
3     13.49       339.31     C        C1     ...               NaN     NaN   
4     12.69        67.79     B        B5     ...               NaN     NaN   

  open_rv_12m  open_rv_24m max_bal_bc all_util total_rev_hi_lim inq_fi  \
0         NaN          NaN        NaN      NaN              NaN    NaN   
1         NaN          NaN        NaN      NaN              NaN    NaN   
2         NaN          NaN        NaN      NaN              NaN    NaN   
3         NaN          NaN        NaN      NaN              NaN    NaN   
4         NaN          NaN        NaN      NaN              NaN    NaN   

  total_cu_tl inq_last_12m  
0         NaN          NaN  
1         NaN          NaN  
2         NaN          NaN  
3         NaN          NaN  
4         NaN          NaN  

[5 rows x 74 columns]

In [5]:
# plot the data info for each feature
row,col = df.shape
for i in range(0,col,10): 
    print(df.iloc[:,i:i+10].head())

        id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501    1296599     5000.0       5000.0           4975.0   36 months   
1  1077430    1314167     2500.0       2500.0           2500.0   60 months   
2  1077175    1313524     2400.0       2400.0           2400.0   36 months   
3  1076863    1277178    10000.0      10000.0          10000.0   36 months   
4  1075358    1311748     3000.0       3000.0           3000.0   60 months   

   int_rate  installment grade sub_grade  
0     10.65       162.87     B        B2  
1     15.27        59.83     C        C4  
2     15.96        84.33     C        C5  
3     13.49       339.31     C        C1  
4     12.69        67.79     B        B5  
                  emp_title emp_length home_ownership  annual_inc  \
0                       NaN  10+ years           RENT     24000.0   
1                     Ryder   < 1 year           RENT     30000.0   
2                       NaN  10+ years           RENT     12252.0 

In [6]:
# check the feature with NAN and set a threshold for the NAN number 
featureCount =  df.isnull().sum().to_dict()
#print(featureCount)
threshold = int(len(df)*0.90)
#print(threshold)
nanfeature = []
for i, j in featureCount.items():
    #print(i,j)
    if j >= threshold:
        nanfeature.append(i)
print('usless feature number is', len(nanfeature))
#print(nanfeature)
# clean the data 
df_loan = df.drop(nanfeature,axis = 1)
# check the size again
print('data size after cleaning the NAN is:', df_new.shape)
print(df_loan.head(2))

usless feature number is 17


NameError: name 'df_new' is not defined

In [9]:
dropCol = ['url','desc']
df_loan.drop(dropCol,axis = 1)
df_loan.shape

(887379, 57)

In [10]:
# split the issue_d to issue_month, issue_year
df_loan['issue_month'], df_loan['issue_year'] = df_loan['issue_d'].str.split('-',1).str
df_loan['last_pymnt_month'], df_loan['last_pymnt_year'] = df_loan['last_pymnt_d'].str.split('-',1).str

In [11]:
# save the data at new csv file
df_loan.to_csv('clean_data.csv')

# summary
1. data size: (887379, 74)
2. useless feature: 19 the useless feature define as 90% of the records are empty, the may not help for furture anaysis and then delete it. Please note: we delay it juse because it is useless for our current analysis, later we still need to go though the data and find any relation ship between these features
3. reformat some features